trees2.java

```diff
@@ -0,0 +1,56 @@
+/**
+ * Definition for a binary tree node.
+ * public class TreeNode {
+ *     int val;
+ *     TreeNode left;
+ *     TreeNode right;
+ *     TreeNode() {}
+ *     TreeNode(int val) { this.val = val; }
+ *     TreeNode(int val, TreeNode left, TreeNode right) {
+ *         this.val = val;
+ *         this.left = left;
+ *         this.right = right;
+ *     }
+ * }
+ */
+
+
+//Time complextiy = O(N square)
+//Space Complexity =O(N)
```

Comment:
Time here should be O(n) and Space O(n^2). Every node is visited only once but a new list on length n is created at n nodes, hence n^2.